Undefined symbols for architecture x86_64: "torch::jit::Value::setUniqueName(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&)"

Mac OS X builds failing this way.

```
21:47:50 Undefined symbols for architecture x86_64:
21:47:50   "torch::jit::Value::setUniqueName(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&)", referenced from:
21:47:50       torch::jit::Block::addInput(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >) in test_jit.cpp.o
21:47:50       torch::jit::Graph::freeValue(torch::jit::Value*) in test_jit.cpp.o
21:47:50 ld: symbol(s) not found for architecture x86_64
21:47:50 clang: error: linker command failed with exit code 1 (use -v to see invocation)
21:47:51 [81/81] Linking CXX executable bin/test_api
```

This appears to be correlated with worker-macos-high-sierra-2 and worker-macos-high-sierra-16.

 https://ci.pytorch.org/jenkins/job/pytorch-builds/job/pytorch-macos-10.13-py3-test2/1078//console https://ci.pytorch.org/jenkins/job/pytorch-builds/job/pytorch-macos-10.13-py3-test2/1073//console  https://ci.pytorch.org/jenkins/job/pytorch-builds/job/pytorch-macos-10.13-py3-test2/1082//consoleFull  https://ci.pytorch.org/jenkins/job/pytorch-builds/job/pytorch-macos-10.13-py3-test2/1070//consoleFull  https://ci.pytorch.org/jenkins/job/pytorch-builds/job/pytorch-macos-10.13-py3-test2/1074//consoleFull  https://ci.pytorch.org/jenkins/job/pytorch-builds/job/pytorch-macos-10.13-py3-test2/1070//consoleFull  https://ci.pytorch.org/jenkins/job/pytorch-builds/job/pytorch-macos-10.13-py3-test2/1071//consoleFull

--- a/.jenkins/pytorch/macos-test.sh
+++ b/.jenkins/pytorch/macos-test.sh
@@ -22,6 +22,13 @@ git submodule update --init --recursive
 export CMAKE_PREFIX_PATH=${PYTORCH_ENV_DIR}/miniconda3/
 
 # Test PyTorch
+if [[ "${JOB_BASE_NAME}" == *cuda9.2* ]]; then
+  # Eigen gives "explicit specialization of class must precede its first use" error
+  # when compiling with Xcode 9.1 toolchain, so we have to use Xcode 8.2 toolchain instead.
+  export DEVELOPER_DIR=/Library/Developer/CommandLineTools
+else
+  export DEVELOPER_DIR=/Applications/Xcode9.app/Contents/Developer
+fi
 export MACOSX_DEPLOYMENT_TARGET=10.9
 export CXX=clang++
 export CC=clang

